### ISSUE TITLE
FeatureRequest: Add PRESERVE_COMMENTS feature for SnakeYAML > 2.0

### ISSUE THREAD

**Issue by bmarwell** [FasterXML/jackson-dataformats-text#491]
Hi,

## Description

currently I am using Jackson with dataformat-yaml to sort my yaml files by key:

```
 final YAMLFactory yamlFactory = new YAMLFactory()
        .disable(YAMLGenerator.Feature.SPLIT_LINES)
        .disable(YAMLGenerator.Feature.WRITE_DOC_START_MARKER)
        ;
    this.objectMapper = new ObjectMapper(yamlFactory)
        .registerModule(new Jdk8Module())
        .setDefaultPrettyPrinter(new DefaultPrettyPrinter())
        .configure(SerializationFeature.INDENT_OUTPUT, true)
        .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
        .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, orderMapEntriesByKeys)
    ;
```

Now, comments will get lost regardless of which method I use to read and write the document.

## SnakeYAML features

However, with SnakeYAML alone you could do something like this:

1. Create a new Reader/Writer which preserves comments: https://stackoverflow.com/a/76975195/1549977
2. Create a custom Representer which sorts keys: https://stackoverflow.com/a/63095199/1549977

## Wishlist

Ideally, a comment above a key in the same indentation level stays with the key. If it is the same indentation level as the previous element, keep it below that one.

Can this be done in Jackson with dataformat-yaml?

**Comment by cowtowncoder** [comment]
No, this cannot be done currently.

But this has come up multiple times, and I think we have some existing issues.
One big problem is that it is very difficult to preserve comment content in `JsonNode` (or similar) since they can exist anywhere within content model. So while adding ability read/write comments at low level (parser/generator) is quite possible, we have nothing to do that at higher level(s).

So this is the problem to resolve.

Also affects XML, fwtw, and likely other formats (TOML at least? even some binary formats; albeit nothing with schema).

**Comment by bmarwell** [comment]
Yes, I suspected this much. I will wonder how to deal with that in tools that consume and auto-sort YAML, maybe do replacements after saving files. Thanks for your quick confirmation! 👍🏻

**Comment by yawkat** [comment]
#36 #254

**Comment by sdavids** [comment]
Another use case is [license headers](https://reuse.software):

```yaml
---
# SPDX-FileCopyrightText: © 2026 Sebastian Davids <sdavids@gmx.de>
# SPDX-License-Identifier: Apache-2.0

key: value
```
